lib/arm/arm_lifter.ml

```diff
@@ -322,20 +322,22 @@ let lift_bits mem ops (insn : bits_insn ) =
     let s = exp_of_op src in
     let i24 = int32 24 in
     let i8 = int32 8 in
-    let umask = int32 0xff0000 in
-    let lmask = int32 0xff00 in
     let rev =
       let open Bil in
       s              lsl i24 lor
       s              lsr i24 lor
-      (s land umask) lsr i8  lor
-      (s land lmask) lsl i8
+      (extract 23 16 s) lsr i8  lor
```

Comment:
see the comment about REV16, it is also applicable here. And please, also just use a concatenation of 4 bytes, obtained with `extract`.